Remove loading of JQuery UI CSS from CDN

theRealRizeo opened the pull request:
JQuery UI CSS was loading from an external CDN. This PR aims to include only required styling for elements used and deprecates the function `wpbdp_enqueue_jquery_ui_style()`.
Pages affected will be :

- Editing the `Expire Date` in a listing admin view
- Editing a listing in the frontend that has a `Date` Field
- CSV Import with `Select a default user to be used if the username column is not present in the CSV file.` checked for the user list autocomplete
- Modal delete prompt in `Form Fields` page

New styling:

Listing Expire Date admin:
![image](https://user-images.githubusercontent.com/121492/158402180-e78ede73-7d0a-437c-8545-d8d3836ad26b.png)

Listing Date Field frontend:
![image](https://user-images.githubusercontent.com/121492/158402293-7e72cf9f-a681-4a21-a8e6-6ca9d176dec0.png)

CSV Import:
![image](https://user-images.githubusercontent.com/121492/158402042-71c72d8a-c0ca-44da-93ae-a6cfc13a76de.png)

Modal Delete Prompt:
![image](https://user-images.githubusercontent.com/121492/158402394-d9f9dd69-ab9b-4f89-bd8a-aba652e0d992.png)

stephywells reviewed (changes requested):
It looks like this could still use some work. It's not looking great. We can just copy the css from Formidable (it's only 49 lines). The colors for the front-end should be based on the button color setting in BD.

Here's how it looks with Formidable installed:
<img width="423" alt="Screen Shot 2022-03-15 at 3 12 29 PM" src="https://user-images.githubusercontent.com/1116876/158474055-8b80f710-f8b0-46d4-9d27-6a811159806c.png">

Here's the way it should look (we don't have to switch to dropdowns though):
![date_ui-calendar](https://user-images.githubusercontent.com/1116876/158474129-efe37836-6da3-4f93-93b6-17592383a441.png)

Also, I think it would be better if we address the to do in there, mentioning that jquery-datepicker-ui should only be loaded when it's needed. We could use the wpbdp_enqueue_jquery_ui_style() function for this, and use wp_enqueue_script( 'jquery-ui-datepicker' ); in that function.

Also, can you help me understand what the new javascript is for in this PR?

theRealRizeo commented:
The new JS mainly is for the `Today` on click to populate the input field with the current date

theRealRizeo commented:
Formidable seems to load a custom jQuery UI lib https://github.com/Strategy11/formidable-pro/tree/master/css/ui-lightness
I have added the lib and made some CSS changes

Front Field:
![image](https://user-images.githubusercontent.com/121492/158562624-7687c473-d3a4-4185-8841-8e87a2997618.png)

Admin backend:
![image](https://user-images.githubusercontent.com/121492/158562741-748a2c3a-3101-4362-a1be-ca755bc341b2.png)

theRealRizeo commented:
@stephywells Thanks for the feedback :) I removed the Jquery UI lib and used the CSS from Formidable. There are CSS elements not fully styled with the Formidable CSS which I added in the `jquery-ui-styles.less` file

garretlaxton commented:
Looks good, but since we are using the Formidable code here it looks like the 2 are causing some minor front-end issues when both are installed.

When I activate Formidable, the dates on the front-end have some border issues on the bottom and right side.
![Screenshot 2022-03-17 at 08-54-09 Add Listing – 0xB9 Sec](https://user-images.githubusercontent.com/16290508/158814438-9c89a070-cfd3-4205-8260-eb21bb82cdbe.png)

And on the receipt page, the CSS for the `<tr>` & `<td>` tags are included in the table here.
![Screenshot 2022-03-17 at 08-55-22 Add Listing – 0xB9 Sec](https://user-images.githubusercontent.com/16290508/158814708-2ee0efbb-35f0-451e-8511-3067e649e014.png)

theRealRizeo commented:
Thanks @garretlaxton I'll add some conditional checks to see if the CSS is loaded :)

theRealRizeo commented:
@garretlaxton pushed a change to handle this. I do not have a Formidable license so I can't check the interaction well

garretlaxton commented:
I'm seeing `margin: 0 0 0.4em;` in the `.ui-datepicker table` class of this file here [/assets/css/jquery-ui-styles.min.css](https://github.com/Strategy11/business-directory-plugin/blob/f9e7ead07e87998d092adc0fae70b74b43b70f70/assets/css/jquery-ui-styles.min.css) causing the bottom margin issue.

theRealRizeo commented:
Calendar in backend has been adjusted to look better

![image](https://user-images.githubusercontent.com/121492/158958784-8b9fa5b0-cc0a-426d-bd74-6113c2a774fb.png)

Padding issue fixed in frontend, thanks @garretlaxton 

![image](https://user-images.githubusercontent.com/121492/158958861-295ccbc5-b18d-46a0-864d-e5ccc60c02d6.png)

garretlaxton reviewed (approved):
Looks great!

stephywells commented:
@garretlaxton would you mind taking one more look over this? I've made some updates so the datepicker script is only loaded on the front-end when there's a date field in the form.

garretlaxton commented:
@stephywells Looks good. I only see `ui-datepicker` loaded in when a date field is in a form on the frontend.
